### PR TITLE
Fast projection check

### PR DESCRIPTION
**Pull request by shreeramsenthi**
Based on previous feedback from @BadgerOnABike and @cptcitrus , there's no need to reproject the whole fuels map to ensure that R can extract lat-long from the raster. Also, it seems `project(x, "+proj=longlat")` is being deprecated, so I swapped that out as well.

I'm afraid I also not set up to test this on real data, so I'd appreciate some help testing if you guys have the time.

If this looks good, there are probably parallel changes to be made in the add on packages.

(This should probably also be sqaushed merge, sorry!)

### CODE REVIEW

**Comment by BadgerOnABike** [comment]
Also added a catch for geographic coordinate systems.

**Comment by BadgerOnABike** [comment]
Take a peek @shreeramsenthi changed a bit, added another layer of check. This should do what we're after now I think.

File: src/conditions.R [hunk 1]
```diff
@@ -63,7 +63,8 @@ if(nrow(WeatherZoneTable) == 0)
 ## Check raster inputs for consistency ----
 
 # Ensure fuels crs can be converted to Lat / Long
-tryCatch(fuelsRaster %>% project("+proj=longlat"), error = function(e) stop("Error parsing provided Fuels map. Cannot calculate Latitude and Longitude from provided Fuels map, please check CRS."))
+if(fuelsRaster %>% is.lonlat){stop("Incorrect coordinate system. Projected coordinate system required, please reproject your grids.")}
+tryCatch(fuelsRaster %>% project("epsg:4326"), error = function(e) stop("Error parsing provided Fuels map. Cannot calculate Latitude and Longitude from provided Fuels map, please check CRS."))
```
Comment (shreeramsenthi) [inline src/conditions.R:67]: What if we made a small temp raster inline and assigned it the same CRS so we don't have to reproject the whole fuels raster?

```suggestion
tryCatch(rast(crs = crs(fuelsRaster)) %>% project("epsg:4326"), error = function(e) stop("Error parsing provided Fuels map. Cannot calculate Latitude and Longitude from provided Fuels map, please check CRS."))
```

File: src/ignitions.R [hunk 2]
```diff
@@ -64,7 +64,8 @@ if(nrow(ResampleOption) == 0) {
 ## Check raster inputs for consistency ----
 
 # Ensure fuels crs can be converted to Lat / Long
-tryCatch(fuelsRaster %>% project("+proj=longlat"), error = function(e) stop("Error parsing provided Fuels map. Cannot calculate Latitude and Longitude from provided Fuels map, please check CRS."))
+if(fuelsRaster %>% is.lonlat){stop("Incorrect coordinate system. Projected coordinate system required, please reproject your grids.")}
+tryCatch(fuelsRaster %>% project("epsg:4326"), error = function(e) stop("Error parsing provided Fuels map. Cannot calculate Latitude and Longitude from provided Fuels map, please check CRS."))
```
Comment (shreeramsenthi) [inline src/ignitions.R:68]:
```suggestion
tryCatch(rast(crs = crs(fuelsRaster)) %>% project("epsg:4326"), error = function(e) stop("Error parsing provided Fuels map. Cannot calculate Latitude and Longitude from provided Fuels map, please check CRS."))
```

**Review by shreeramsenthi — COMMENTED** [review]
Makes sense to me, I proposed one other possible speed up to the CRS check, but maybe I'm just splitting hairs at this point.

**Pull request merged**